Directory.Packages.props

```diff
@@ -141,6 +141,7 @@
     <PackageVersion Include="OpenTelemetry.Instrumentation.Runtime" Version="1.9.0" />
     <!-- build dependencies -->
     <PackageVersion Include="MicroBuild.Plugins.SwixBuild.Dotnet" Version="1.1.87-gba258badda" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.11.0" />
```

Comment:
We are going to have to be careful on versioning now. Previously we only referenced Roslyn in our internal build for the Configuration JSON schema generator. So we always updated the version, because it was only used for our build. That's why the existing CodeAnalysis dependences were listed under `<!-- build dependencies -->`.

But now that we are building analyzers that we give out to customers, we will need to be more thoughtful when this version gets updated.

I suggest at least splitting this new PackageVersion into a separate section that isn't under `<!-- build dependencies -->`. I also think it might make sense for us to not use CPM for this dependency and instead have an explicit MSBuild property for the version, and use `PackageOverride` in the analyzer. (Although I've never attempted to build an analyzer in a repo that uses CPM and dependabot before, so there might be better approaches.)

Reply:
I'll try detaching the analyzer project from CPM and setting it specifically for the reasons you point out.